### PR TITLE
Add Dialog.Content closeOnPointerDownOutside prop

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Allow to close dialog on click outside via prop. Use it in information dialogs about tier coverage and deductible in Price Calculator page

Since many of current dialogs are modals which we don't want to close accidentally, prop defaults to false

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4904?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4904** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4904?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4903** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4903?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -10,11 +11,25 @@ type ContentProps = {
...
   frostedOverlay?: boolean
   centerContent?: boolean
+  closeOnPointerDownOutside?: boolean
+  asChild?: boolean
 }
 
 const Content = (props: ContentProps) => {
   const handleClose = () => props.onClose?.()
 
+  const handleInteractOutside = useCallback(
```
Comment (Youakeem) [inline packages/ui/src/components/Dialog/Dialog.tsx:21]: I am not sure if `useCallback` is needed

Comment (alebedev) [reply packages/ui/src/components/Dialog/Dialog.tsx:21]: Good point, we did not have it

File: packages/ui/src/components/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -10,11 +11,21 @@ type ContentProps = {
...
+  asChild?: boolean
 }
 
 const Content = (props: ContentProps) => {
   const handleClose = () => props.onClose?.()
 
+  const handleInteractOutside = (
+    event: Parameters<NonNullable<DialogContentProps['onInteractOutside']>>[0],
+  ) => {
+    if (event.type !== 'dismissableLayer.pointerDownOutside' || !props.closeOnPointerDownOutside) {
```
Comment (Youakeem) [inline packages/ui/src/components/Dialog/Dialog.tsx:24]: I am not sure about defaulting to false, I feel like the natural behaviour is to click outside to close, are we sure we want to have this as the default behaviour?

On a side note: Don't think we really care about what the event is called we can omit that from the prop name.

Comment (alebedev) [reply packages/ui/src/components/Dialog/Dialog.tsx:24]: If we had no existing usages, I would agree. But allowing existing dialogs like BankID to close by random click outside might break things in unexpected way. So, making it opt-in is safer

Comment (Youakeem) [reply packages/ui/src/components/Dialog/Dialog.tsx:24]: Can't we op into the "modal" behaviour instead?

Comment (alebedev) [reply packages/ui/src/components/Dialog/Dialog.tsx:24]: Do you mean `Dialog.Root.modal` (https://www.radix-ui.com/primitives/docs/components/dialog#root). It's already on by default and does not affect mouse interaction, unfortunately - only keyboard focus

So, I'd love to use built-in library feature for this, but don't see one that does what we want

**Review by Youakeem — APPROVED** [review]

**Comment by alebedev** [comment]
### Merge activity

* **Oct 29, 6:00 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4904).

**Pull request merged**